hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContract.php

```diff
@@ -193,21 +194,26 @@ protected static function getContractDates($contracts) {
     // If there are two (or more) Contracts starting on the same day
     // then we pick only the one with latest End date.
     foreach ($contracts['values'] as $contract) {
-      $details = civicrm_api3('HRJobDetails', 'getsingle', array(
+      $details = civicrm_api3('HRJobDetails', 'get', array(
         'sequential' => 1,
         'jobcontract_id' => (int)$contract['id'],
+        'period_start_date' => array('<=' => $date),
       ));
-      if (empty($details['period_end_date'])) {
-        $dates[$details['period_start_date']] = null;
-        break;
+      if (empty($details['values'])) {
+        continue;
+      }
+      $detailsValues = CRM_Utils_Array::first($details['values']);
+      if (empty($detailsValues['period_end_date'])) {
+        $dates[$detailsValues['period_start_date']] = null;
+        continue;
       }
-      if (!empty($dates[$details['period_start_date']])) {
-        if ($details['period_end_date'] > $dates[$details['period_end_date']]) {
-          $dates[$details['period_start_date']] = $details['period_end_date'];
+      if (!empty($dates[$detailsValues['period_start_date']])) {
```

Comment:
this can be removed since currently we cannot have two contracts with the same "start_date" :
```php
      if (!empty($dates[$detailsValues['period_start_date']])) {
        if ($detailsValues['period_end_date'] > $dates[$detailsValues['period_start_date']]) {
          $dates[$detailsValues['period_start_date']] = $detailsValues['period_end_date'];
        }
        continue;
      }
```